erpnext/projects/doctype/task/task.py

```diff
@@ -198,12 +199,21 @@ def has_webform_permission(self):
 		if project_user:
 			return True
 
-	def update_status(self):
-		if self.status not in ('Cancelled', 'Completed') and self.exp_end_date:
-			from datetime import datetime
-			if self.exp_end_date < datetime.now().date():
-				self.db_set('status', 'Overdue', update_modified=False)
-				self.update_project()
+	def set_is_overdue(self):
```

Comment:
- Should be a setter, update db only if argument provided
- Logic should be based on default 0
- Should only affect is_overdue, no need to call `update_project` or `notify_update`